laterpay/signing.py

```diff
@@ -62,33 +79,53 @@ def normalise_param_structure(params):
     """
     Canonicalise representation of key-value data with non-unique keys.
 
-    Request parameter dictionaries are handled in different ways in different libraries,
-    this function is required to ensure we always have something of the format:
-
-       {  key:  [ value1, value2... ] }
+    Request parameter dictionaries are handled in different ways in different
+    libraries. This function is required to ensure we always have something of
+    the format::
+
+        {
+            'key1': ['value1', 'value2'],
+            'key2': ['value3'],
+        }
+
+    :param dict, list, tuple, furl.omdict1D.omdict params: The parameter
+        structure to normalize. Can be either a ``dict``, ``list``, ``tuple``
+        or ``furl.omdict1D.omdict``. The following formats are allowed::
+
+            # A dictionary with key-value or key-values mappings
+            {
+                'key1': 'value',
+                'key2': ['value1', 'value2'],
+                'key3': ('value1', 'value2'),
+            }
+
+            # A list (or tuple, can be used interchangeably)
+            [
+                ['key1', 'value1'],
+                ['key1', 'value2'],
+                ['key2', ['value1', 'value2']],
+            ]
 
     """
-    out = {}
+    if isinstance(params, dict):
```

Comment:
Maybe this part would be better if using `collections.abc`? It might give some flexibility to the caller, whoever they are.